### DISCUSSION TITLE
Why is "add_question_sql" storing vectors as question+sql? SQL isn't natural language, so it wouldn't affect the semantic matching of input queries?

### DISCUSSION THREAD

**Discussion by qingwu11** [vanna-ai/vanna#452] [Q&A]
![ques+sql](https://github.com/vanna-ai/vanna/assets/41281228/86408651-cbd9-4cdd-bf2f-210b1e182dd8)
Why is "add_question_sql" storing vectors as question+sql? SQL isn't natural language, so it wouldn't affect the semantic matching of input queries?

**Comment by erickrribeiro** [comment]
Yes, @qingwu11. I also wondered the same thing when I saw how this id is created. I think that only the `question` field should be used to generate the unique code and not the `question+sql`

**Comment by zainhoda** [comment]
In our testing we actually did find better search by embedding the SQL query particularly if it contains specific terminology that may not be contained in the question.

However, to @erickrribeiro 's point -- I think that the ID should only be based on the question

**Reply by geraldFZh** [reply]
were the tests all conducted in English? like English questions only, no non-English questions included

**Comment by erickrribeiro** [comment]
Thanks @zainhoda for considering the ID suggestion based on the question. Continuing with the subject, I separated three advantages of using the ID in this way:

1. Vanna will not have two SQLs pointing to the same question. When two sqls point to the same question there may possible that one of them is incorrect, when this occurs it creates ambiguity in the system.

2.  ID based on question will facilitate the process of updating the SQL of an existing document. For example, in ChromaDb you can use [upsert](https://docs.trychroma.com/reference/py-collection#upsert)

3. It will be easy to create a feature to update a document

![image](https://github.com/vanna-ai/vanna/assets/5834111/7085e143-42fd-485c-a49c-0fe813a339b6)

**Comment by watertianyi** [comment]
@erickrribeiro Is training these ddl, sql and document files mainly for offline vector storage operations? Do these files need to be defined according to their own databases? Is each input query question converted into a vector and then the similarity calculated with the three file vectors in the vector library? Then combine it into prompt and send it to llm? Is this the workflow?
